onnx/defs/tensor/defs.cc

```diff
@@ -1026,6 +1052,49 @@ ONNX_OPERATOR_SET_SCHEMA(
                 ->mutable_dim((int)axis)
                 ->set_dim_value(temp);
           }
+        })
+        .PartialDataPropagationFunction([](DataPropagationContext& ctx) {
+          const auto input_data = ctx.getInputData(0);
+          const auto starts = ctx.getInputData(1);
+          const auto ends = ctx.getInputData(2);
+          const auto axes = ctx.getNumInputs() >= 4 ? ctx.getInputData(3) : nullptr;
+          const auto steps = ctx.getNumInputs() >= 5 ? ctx.getInputData(4) : nullptr;
+
+          if (input_data == nullptr || starts == nullptr || ends == nullptr) {
+            return;
+          }
+          if (starts->dim_size() != ends->dim_size()) {
+            fail_shape_inference("Input rank for starts and ends should be the same: (",
+            starts->dim_size(), ") vs (", ends->dim_size(), ").");
+          }
+          // Only supports axis = 0 since the data comes from Shape
+          if((axes == nullptr || (axes->dim_size() == 1 && axes->dim(0).dim_value() == 0))
+            && starts->dim_size () == 1 && ends->dim_size() == 1) {
+            auto start = starts->dim(0).dim_value();
+            auto end = ends->dim(0).dim_value();
+            int64_t step = 1; // Default step is 1
+            if (steps != nullptr) {
+              if (steps->dim_size() != 1) {
+                return;
+              }
+              step = steps->dim(0).dim_value();
+            }
+
+            processSliceInputs(input_data->dim_size(), start, end, step);
+            TensorShapeProto tsp;
+            if (start < end) {
```

Comment:
nit: using ```step > 0``` here may be slightly better and preferable. (Similarly in shape inference method above.)